src/Compilers/Core/Rebuild/CompilationOptionsReader.cs

```diff
@@ -276,7 +276,7 @@ where PdbReader.GetGuid(cdi.Kind) == EmbeddedSourceGuid
                 var content = new byte[length];
                 Marshal.Copy(new IntPtr(contentPtr), content, 0, length);
 
-                var isPublic = (resource.Attributes & ManifestResourceAttributes.Public) != 0;
+                var isPublic = (resource.Attributes & ManifestResourceAttributes.VisibilityMask) == ManifestResourceAttributes.Public;
```

Comment:
This doesn't appear to be equivalent. Yes, the only 2 flags are `Public` and `Private`, so it's arguable that `Public | Private` shouldn't be true here, but I'd prefer to keep this as is.

Reply:
why do you think this is not equivalent? the mask nulls the other bits (which don't exist in this enum ..yet?) so we can then directly compare the entry.

I'm not saying this is nice but it's the pattern these *Mask APIs are using

Reply:
It's not equivalent exactly as I said: if `resource.Attributes` is `Public | Private`, this will now be false, where the existing code would say true.

Reply:
ah I misread what you said, you meant that before `Public | Private` returned true here while now it doesn't.

that's true but I'd find it kind of strange for this usage